packages/common/src/common.ts

```diff
@@ -466,14 +487,85 @@ export class Common {
         }
       }
     }
-    this._eips = eips
+  }
+
+  /**
+   * Internal helper for _buildParamsCache()
+   */
+  protected _mergeWithParamsCache(params: HardforkConfig | EIPConfig) {
+    this._paramsCache['gasConfig'] = {
+      ...this._paramsCache['gasConfig'],
+      ...params['gasConfig'],
+    }
+    this._paramsCache['gasPrices'] = {
+      ...this._paramsCache['gasPrices'],
+      ...params['gasPrices'],
+    }
+    this._paramsCache['pow'] = {
+      ...this._paramsCache['pow'],
+      ...params['pow'],
+    }
+    this._paramsCache['sharding'] = {
+      ...this._paramsCache['sharding'],
+      ...params['sharding'],
+    }
+    this._paramsCache['vm'] = {
+      ...this._paramsCache['vm'],
+      ...params['vm'],
+    }
+  }
+
+  /**
+   * Build up a cache for all parameter values for the current HF and all activated EIPs
+   */
+  protected _buildParamsCache() {
+    this._paramsCache = {}
+    // Iterate through all hardforks up to hardfork set
+    const hardfork = this.hardfork()
+    for (const hfChanges of this.HARDFORK_CHANGES) {
+      // EIP-referencing HF config (e.g. for berlin)
+      if ('eips' in hfChanges[1]) {
+        const hfEIPs = hfChanges[1]['eips']
+        for (const eip of hfEIPs!) {
+          if (!(eip in EIPs)) {
+            throw new Error(`${eip} not supported`)
+          }
+
+          this._mergeWithParamsCache(EIPs[eip])
+        }
+        // Parameter-inlining HF config (e.g. for istanbul)
+      } else {
+        this._mergeWithParamsCache(hfChanges[1])
+      }
+      if (hfChanges[0] === hardfork) break
+    }
+    // Iterate through all additionally activated EIPs
+    for (const eip of this._eips) {
+      if (!(eip in EIPs)) {
+        throw new Error(`${eip} not supported`)
+      }
+
+      this._mergeWithParamsCache(EIPs[eip])
+    }
+  }
+
+  protected _buildActivatedEIPsCache() {
+    this._activatedEIPsCache = []
+
+    for (const hfChanges of this.HARDFORK_CHANGES) {
+      const hf = hfChanges[1]
+      if (this.gteHardfork(hf['name']) && 'eips' in hf) {
+        this._activatedEIPsCache = this._activatedEIPsCache.concat(hf['eips'] as number[])
+      }
+    }
+    this._activatedEIPsCache = this._activatedEIPsCache.concat(this._eips)
```

Comment:
For even more efficiency for activated EIPs we might want to use a `Set` instead of an object.

Reply:
(this is not necessary for this PR)